app/controllers/application_controller.rb

```diff
@@ -69,7 +69,11 @@ def render_unauthorized
   end
 
   def saml_settings
-    SAML::SettingsService.instance.saml_settings
+    settings = SAML::SettingsService.instance.saml_settings
+    # TODO: 'level' should be its own class with proper validation
+    level = LOA::MAPPING.invert[params[:level]&.to_i]
+    settings.authn_context = level || LOA::MAPPING.invert[1]
```

Comment:
We could still raise an error here if there is a bad level provided if that is the desired functionality. Also, you can remove the `nil` check, `nil.to_i = 0`